spring-integration-websocket/src/main/java/org/springframework/integration/websocket/inbound/WebSocketInboundChannelAdapter.java

```diff
@@ -374,7 +379,10 @@ private void produceConnectAckMessage(Message<?> message, SimpMessageHeaderAcces
 	}
 
 	private void produceMessage(Message<?> message, SimpMessageHeaderAccessor headerAccessor) {
-		Object payload = this.messageConverter.fromMessage(message, this.payloadType.get());
+		Class<?> targetType = this.payloadType.get();
+		Assert.notNull(targetType, "payloadType must not be null");
```

Comment:
This is not correct: the value of that `AtomicReference` is never null.
Actually, I'm thinking if it has to be an `AtomicReference` at all.
Can you experiment if that could be just plain `Class<?>` property?

Reply:
Agreed. No need for `AtomicReference` since this isn't even expression.